pkg/asset/machines/clusterapi.go

```diff
@@ -167,7 +166,7 @@ func (c *ClusterAPI) Generate(ctx context.Context, dependencies asset.Parents) e
 			return fmt.Errorf("failed to create CAPA tags from UserTags: %w", err)
 		}
 
-		publicOnlySubnets := os.Getenv("OPENSHIFT_INSTALL_AWS_PUBLIC_ONLY") != ""
+		publicOnlySubnets := awstypes.IsPublicOnlySubnetsEnabled()
```

Comment:
nit: Either here or in the install config validation can we add a Warning that this feature is enabled and not recommended (similar to overriding the image)? This way no validation is used (like originally suggested), but we know when the feature was set.

Reply:
Like this? https://github.com/openshift/installer/pull/8883/files#diff-bab1b8072f369df53e27d65607145fb93caabd3a86fdefbcafb176e823726069R55

Reply:
yes, apparently I missed that !